### PR TITLE
Move sys-libs/pam to portage-stable

### PR DESCRIPTION
**Pull request by krnowak**
Blocked on #1685.

Add some compat symlink creation to coreos-base/misc-files. Move our code modification to a user patch. Updates the package to 1.5.3-r1.

TODO:
 - CI (after the base PR is merged)
 - check if our modification for using setuid binary instead of adding DAC_OVERRIDE cap is still necessary.

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [ ] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

File: sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/profile.bashrc [hunk 1]
```diff
@@ -134,6 +134,45 @@ cros_pre_pkg_postinst_no_modifications_of_users() {
...
+            entry_basename=${etc_entry##*/}
+            if [[ -d ${etc_entry} ]]; then
+                dir_pairs=( "${etc_dir}/${entry_basename}" "${usr_dir}/${entry_basename}" "${dir_pairs[@]}")
+            else
+                dodir "${usr_dir}"
+                mv "${etc_entry}" "${ED}${usr_dir}/${entry_basename}" || die
+                einfo "Moved ${etc_dir}/${entry_basename} to ${usr_dir}/${entry_basename}"
+            fi
+        done
+    done
```
Comment (chewi) [inline sdk_container/src/third_party/coreos-overlay/profiles/coreos/base/profile.bashrc]: This hurts my head. Would it work as a recursive function?

**Comment by krnowak** [comment]
Alright, rebased and the CI mostly passed (brightbox failed due to timeout in preparing the image).

File: sdk_container/src/third_party/coreos-overlay/coreos/user-patches/sys-auth/pambase/0001-Reorganize-the-login-sessions.patch [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
++session		required	pam_selinux.so open
+diff --git a/templates/system-session.tpl b/templates/system-session.tpl
+index 150061f..690396f 100644
+--- a/templates/system-session.tpl
++++ b/templates/system-session.tpl
+@@ -12,6 +12,7 @@ session		[success=1 default=ignore]	pam_krb5.so {{ debug }} ignore_root try_firs
+ session		[success=1 default=ignore]	pam_systemd_home.so
+ {% endif %}
+ 
++session		optional	pam_umask.so {% if debug %}debug{% else %}silent{% endif +%}
```
[the remark this answers is not in the document]
Comment (krnowak) [reply sdk_container/src/third_party/coreos-overlay/coreos/user-patches/sys-auth/pambase/0001-Reorganize-the-login-sessions.patch]: Uh, ok, I think I need to make some clarifications then.

Systemd (upstream) provides a systemd-user PAM config (that gets installed into /usr/lib/pam.d) - it's pretty much a standalone thing. I think Fedora uses it as is.

Gentoo also provides a systemd-user PAM config (that gets installed in /etc/pam.d). I think Gentoo is providing their own config, mostly to hook it into auth provided by sys-auth/pambase. But I think pambase was not using some of pam modules that were added into upstream's systemd-user PAM config over time, and Gentoo's systemd-user wasn't using them either. I don't know exactly why, but it's probably either "slipped through the cracks" or it was the expectation that pambase will add those things. I think it was pam_umask, pam_keyinit and pam_namespace modules that were missing. So to answer your question about pam_umask - it looked like a good place to put it.

So my pambase patch (the first of two, second is just about some Flatcar-specific modifications) is about reworking the config files in pambase slightly, so that Gentoo's systemd-user can just use system-login and, optionally (for USE=selinux) system-selinux configs for the session stuff. Possibly systemd-run0 config too. And also to add the missing modules I mentioned earlier. I tried to kinda-sorta follow what Fedora is doing, but obviously Fedora and Gentoo configs differ too much and it difficult to follow these things as all that stuff is a tangled horrible mess anyway. "Hard to follow how these all fit together" as you said.

I thought that the first pambase patch is something I could try upstreaming, and then systemd-user PAM config in Gentoo could be updated to be more or less what we do in the post_src_install hook.

Another way of doing things would be to just drop the custom PAM configs and use what upstream systemd provides. But I have noticed that Gentoo often replaces PAM configs provided by packages with their own variants that hook into pambase configs, which is the reason I didn't go that way.
